### PR TITLE
Transformation procs now properly grant/remove languages.

### LINKED ISSUE #29576: Monkeys made by Jungle Fever don't have languages set properly
They speak galactic common and can't understand monkey

### PR DESCRIPTION
**Pull request by harmonyn**
Fixes https://github.com/tgstation/tgstation/issues/29576.

To be clear these are the monkeyize, humanize, etc procs (the same ones in the admin panel). There are two things I do not know what to do about:

- SMALLER THING: I don't know what to do about AI-ize
- LARGER THING: From a design standpoint do we want people to lose/gain galcom and gain the new language every time this happens? If not, what refactor would be better?

also I fixed spelling/grammar shit

### CODE REVIEW

**Comment by optimumtact** [comment]
Is there a binary language or is it still a snowflake comms channel? It's probably fine to just give them galcomm

**Comment by bgobandit** [comment]
I mean in code terms, AI mob and mind structure is... special and as you can see the code for it is different (grant_language doesn't run off mind but mob)

**Comment by YakumoChen** [comment]
its both optimum

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Add galcomm and machinespeak to AIs

**Comment by optimumtact** [comment]
sorry I'm not sure I understand the issue? are there missing helper procs for the AI mob path or something?

**Comment by bgobandit** [comment]
The AI-ize puts the mob into an AI core, because that's what AIs do. This fucks up references to mob. The procs for giving languages are designed to be used on mobs, not minds.

**Comment by Cyberboss** [comment]
@bgobandit

**Comment by Cyberboss** [comment]
Just call `grant_language()` on the . parameter of AIize

File: code/modules/mob/transform_procs.dm [hunk 1]
```diff
@@ -129,6 +129,9 @@
 		if(O.mind.changeling)
 			O.mind.changeling.purchasedpowers += new /obj/effect/proc_holder/changeling/humanform(null)
 
+	//handle languages
+	O.remove_all_languages()
```
Comment (AnturK) [inline code/modules/mob/transform_procs.dm]: Won't this drop faction languages like cultspeak ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/transform_procs.dm (hunk 1 above)
Comment (bgobandit) [inline code/modules/mob/transform_procs.dm]: possibly I need to figure out how to handle it

**Comment by KorPhaeron** [comment]
Stale

**Pull request closed without merging**